src/main/java/org/opensearch/knn/index/query/KNNQueryFactory.java

```diff
@@ -81,10 +97,37 @@ public static Query create(CreateQueryRequest createQueryRequest) {
             log.debug(
                 String.format("Creating Lucene k-NN query with filters for index: %s \"\", field: %s \"\", k: %d", indexName, fieldName, k)
             );
-            return new KnnFloatVectorQuery(fieldName, vector, k, filterQuery);
+            if (VectorDataType.BYTE.equals(vectorDataType)) {
```

Comment:
Also, I see there are duplicated checks. Can we just abstract them in a private function which will give us right Vector Query.